OpenRA.Game/Exts.cs

```diff
@@ -132,6 +132,20 @@ public static bool HasModifier(this Modifiers k, Modifiers mod)
 			return ret;
 		}
 
+		public static V GetOrAdd<K, V>(this SortedDictionary<K, V> d, K k)
+			where V : new()
+		{
+			return d.GetOrAdd(k, _ => new V());
+		}
+
+		public static V GetOrAdd<K, V>(this SortedDictionary<K, V> d, K k, Func<K, V> createFn)
+		{
+			V ret;
+			if (!d.TryGetValue(k, out ret))
+				d.Add(k, ret = createFn(k));
+			return ret;
```

Comment:
If you don't mind, I'd suggest adding a new line above this one here for better readability.